packages/backend/src/rpc/serviceRegistry.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { App, BaseService } from '@h4bff/core';
+import { RPCServiceRegistry } from './serviceRegistry';
+import { RPCDispatcher } from './dispatcher';
+import { Request, Response } from 'express';
+
+describe('RPCServiceRegistry', () => {
+  describe('Registration', () => {
+    it(`should register new RPC service`, () => {
+      let app = new App();
+      let rpcServiceRegistry = app.getSingleton(RPCServiceRegistry);
+      class TestService extends BaseService {}
+      rpcServiceRegistry.add('test', TestService);
+      let testServiceResult = rpcServiceRegistry.get('test');
+      expect(testServiceResult).toEqual(TestService);
+    });
+
+    it(`should not allow to register a service under existing alias`, () => {
+      let app = new App();
+      let rpcServiceRegistry = app.getSingleton(RPCServiceRegistry);
+      class TestService extends BaseService {}
+      rpcServiceRegistry.add('test', TestService);
+      expect(() => rpcServiceRegistry.add('test', TestService)).toThrowError('Namespace test already in use!');
+    });
+  });
+
+  describe('Exist', () => {
+    it(`should assert that a given method exists on a registered service`, () => {
+      let app = new App();
+      let rpcServiceRegistry = app.getSingleton(RPCServiceRegistry);
+      class TestService extends BaseService {
+        test() {
+          return Promise.resolve();
+        }
+      }
+      rpcServiceRegistry.add('test', TestService);
+      expect(rpcServiceRegistry.exists('test', 'test')).toBeTruthy();
```

Comment:
is `exists` part of public API?

Reply:
Yes, it is part of the public API. It is used in the dispatcher as you already assumed. I just wanted to test it in isolation here.

Reply:
Its best to avoid testing internal stuff as much as possible...

Reply:
Ok, I will remove the cases that test the internal logic.